aes/src/lib.rs

```diff
@@ -123,12 +130,26 @@
 )]
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
 #![warn(missing_docs, rust_2018_idioms)]
+#![cfg_attr(
+    all(
+        any(target_arch = "riscv32", target_arch = "riscv64"),
+        target_feature = "zknd",
+        target_feature = "zkne"
+    ),
+    feature(riscv_ext_intrinsics)
+)]
 
 #[cfg(feature = "hazmat")]
 pub mod hazmat;
 
 #[macro_use]
 mod macros;
+#[cfg_attr(any(target_arch = "riscv32", target_arch = "riscv64"),
+// The `soft` AES-192 is used for RISC-V when RVV crypto (Zvkned) is supported
+// but Scalar crypto (Zkn{ed}) is not supported. Most of the rest of the `soft`
```

Comment:
This PR is for scalar crypto support, so it looks like this part is not needed.

Reply:
This is still needed to silence warnings about unused definitions for architectures without `autodetect` support (because `soft` will be unused in that case) but the reasoning behind the comment is confusing now that the vector support has been separated out.

I've changed the conditional to this:

```rust
#[cfg_attr(not(any(
    target_arch = "aarch64",
    target_arch = "x86",
    target_arch = "x86_64",
)),
// Architectures that have hardware-support backends, but that do not use
// `autodetect` -- like RISC-V 64 -- will generate warnings about unused
// definitions in `soft`. We silence them once here in order to avoid cluttering
// `soft` with architecture specific conditionals.
allow(unused))]
mod soft;
```